src/Microsoft.Azure.ServiceBus/Amqp/AmqpConnectionHelper.cs

```diff
@@ -99,6 +99,23 @@ internal class AmqpConnectionHelper
             return tpSettings;
         }
 
+        public static TransportSettings CreateWebSocketTransportSettings(
+            string networkHost,
+            string hostName,
+            int port)
+        {
+            UriBuilder uriBuilder = new UriBuilder(WebSocketConstants.WebSocketSecureScheme, networkHost, port < 0 ? WebSocketConstants.WebSocketSecurePort : port, WebSocketConstants.WebSocketDefaultPath);
```

Comment:
Sorry, just noticed this. If the user has given an out of range port, then he is either trying to fool around the system or has a wrong connection string in the first place. In either of the cases, he will get an exception from the lower layer and it won't harm us. But the plus point is that for the second scenario he might realise something is wrong and double check his connection string. We shouldn't be defaulting to our 443 when he is giving a out of range port on purpose. This check for <0 is useful since if no port details are given, UriBuilder returns a default of -1, and that is when we may have to apply our default port.
So in short, the current code looks right to me

---
In reply to: [134849248](https://github.com/Azure/azure-service-bus-dotnet/pull/288#discussion_r134849248) [](ancestors = 134849248)